### PR TITLE
optimized random value generators

### PR DESCRIPTION
**Pull request by danieldietrich**

### CODE REVIEW

File: javaslang-test/src/main/java/javaslang/test/Gen.java [hunk 1]
```diff
@@ -424,3 +413,27 @@
         return f.apply(this);
     }
 }
+
+final class GenModule {
```
Comment (sputnikci) [inline javaslang-test/src/main/java/javaslang/test/Gen.java]: \[Checkstyle\] ERROR: Utility classes should not have a public or default constructor\.

File: javaslang-test/src/main/java/javaslang/test/Gen.java [hunk 2]
```diff
@@ -424,3 +413,27 @@
...
+     * @return A value generator, choosen according to the given frequencies and the underlying n
+     */
+    static <T> Gen<T> frequency(int n, java.util.Iterator<Tuple2<Integer, Gen<T>>> iter) {
+        do {
+            final Tuple2<Integer, Gen<T>> freqGen = iter.next();
+            final int k = freqGen._1;
+            if (n <= k) {
+                return freqGen._2;
+            } else {
+                n = n - k;
```
Comment (sputnikci) [inline javaslang-test/src/main/java/javaslang/test/Gen.java]: \[Checkstyle\] ERROR: Assignment of parameter 'n' is not allowed\.

Comment (danieldietrich) [inline javaslang-test/src/main/java/javaslang/test/Gen.java]: Ok for me - it is performed on a copy on the stack. Btw - when dealing with immutable objects (like in Javaslang), then nothing evil can happen at all!

File: javaslang-test/src/main/java/javaslang/test/Gen.java (hunk 1 above)
Comment (danieldietrich) [inline javaslang-test/src/main/java/javaslang/test/Gen.java]: Will fix that

File: javaslang-test/src/main/java/javaslang/test/Gen.java [hunk 3]
```diff
@@ -424,3 +413,27 @@
...
```
Comment (sputnikci) [inline javaslang-test/src/main/java/javaslang/test/Gen.java]: \[Checkstyle\] ERROR: Assignment of parameter 'n' is not allowed\.

**Pull request merged**